src/responses/Albany_AbstractResponseFunction.hpp

```diff
@@ -68,6 +68,7 @@ namespace Albany {
       const Teuchos::RCP<const Thyra_Vector>& xdot,
       const Teuchos::RCP<const Thyra_Vector>& xdotdot,
       const Teuchos::Array<ParamVec>& p,
+      int parameter_index,
```

Comment:
I don't understand why you need  the parameter index (o why it was not needed before). Shouldn't return the derivatives for all the parameter in the paramVec p?

Reply:
This is used in:

```
evaluateTangent(const double /*alpha*/,
		const double beta,
		const double /* omega */,
		const double /*current_time*/,
		bool /*sum_derivs*/,
    const Teuchos::RCP<const Thyra_Vector>& /*x*/,
    const Teuchos::RCP<const Thyra_Vector>& /*xdot*/,
    const Teuchos::RCP<const Thyra_Vector>& /*xdotdot*/,
		const Teuchos::Array<ParamVec>& p,
    int parameter_index,
		ParamVec* deriv_p,
    const Teuchos::RCP<const Thyra_MultiVector>& /*Vx*/,
    const Teuchos::RCP<const Thyra_MultiVector>& /*Vxdot*/,
    const Teuchos::RCP<const Thyra_MultiVector>& /*Vxdotdot*/,
    const Teuchos::RCP<const Thyra_MultiVector>& Vp,
    const Teuchos::RCP<Thyra_Vector>& g,
    const Teuchos::RCP<Thyra_MultiVector>& gx,
    const Teuchos::RCP<Thyra_MultiVector>& gp)
{
  // Evaluate response g
  if (!g.is_null()) {
    evaluateResponseImpl(p,*g);
  }
  
  if (!gx.is_null()) {
    gx->assign(0.0);
  }

  if (!gp.is_null()) {
    // dgdp = dg/dp
    SerialDenseVector<int, double> dgdp(n_parameters);
    evaluateGradientImpl (p, dgdp);

    // gp = (dg/dp)^T*Vp
    Thyra::set_ele(0, dgdp(parameter_index), gp->col(0).ptr());
  }
}
```

Because more than one scalar parameter are used in this response and because the order matters (due to the fact that we provide the matrix C and the vector theta_0 in the yaml file with a given order), we need to have a way to know the current index of the current parameter.

Does it make more sense?

Reply:
But you can use `deriv_p`  ( = p[parameter_index]) instead of `p'. In this way you avoid computing `evaluateGradientImpl` for parameters you do not need.